src/test/java/lv/ctco/javaschool/goal/boundary/GoalApiTest.java

```diff
@@ -0,0 +1,249 @@
+    List<Goal> goalList1 = new ArrayList<>();
+    List<Goal> goalList2 = new ArrayList<>();
+    List<GoalDto> goalDtoList = new ArrayList<>();
+    String testLine1, expResult1, testLine4, expResult4;
+
+    @Mock
+    private UserStore userStore;
+    @Mock
+    private GoalStore goalStore;
+    @InjectMocks
+    private GoalApi goalApi;
+
+    @BeforeEach
+    void init() {
+        user1.setUsername("user");
+        user1.setEmail("user@user.com");
+        user1.setId(1L);
+        user1.setPassword("12345");
+        user1.setPhone("12345678");
+
+        user2.setUsername("admin");
+        user2.setEmail("admin@admin.com");
+        user2.setId(2L);
+        user2.setPassword("12345");
+        user2.setPhone("87654321");
+
+        goal.setUser(user1);
+        goal.setGoalMessage("abc");
+        goal.setRegisteredDate(LocalDateTime.now());
+        goal.setDeadlineDate(LocalDate.now().plusDays(1));
+        goal.setId(1L);
+        goal.setTags(null);
+
+        goal2.setUser(user2);
+        goal2.setGoalMessage("cde");
+
+        testLine1 = "I will become a programmer this year!";
+        expResult1 = "programmer";
+        testLine4 = "I will start to learn Java!";
+        expResult4 = "learn Java";
+    }
+
+    @Test
+    @DisplayName("Test getMyGoals(): Current user has No goals returns empty list of dto's")
+    void testGetMyGoals() {
+        when(userStore.getCurrentUser())
+                .thenReturn(user1);
+        when(goalStore.getGoalsListFor(user1))
+                .thenReturn(goalList1);
+        assertEquals(goalDtoList, goalApi.getMyGoals());
+    }
+
+    @Test
+    @DisplayName("Test getMyGoals(): Current user has goals returns list of dto's")
+    void testGetMyGoals2() {
+        goalList1.add(goal);
+        when(userStore.getCurrentUser())
+                .thenReturn(user1);
+        when(goalStore.getGoalsListFor(user1))
+                .thenReturn(goalList1);
+        assertEquals(goal.getUser().getUsername(), goalApi.getMyGoals().get(0).getUsername());
+        assertEquals(goal.getGoalMessage(), goalApi.getMyGoals().get(0).getGoalMessage());
+        assertEquals(goal.getId(), (Long) goalApi.getMyGoals().get(0).getId());
+    }
+
+    @Test
+    @DisplayName("Test getGoalById(): returns dto of goal by id")
+    void testGetGoalById2() {
+        when(goalStore.getGoalById((long) 1))
+                .thenReturn(java.util.Optional.empty());
+        assertEquals(GoalDto.class, goalApi.getGoalDtoByGoalId((long) 1).getClass());
+        assertEquals(null, goalApi.getGoalDtoByGoalId((long) 1).getGoalMessage());
+        assertEquals(null, goalApi.getGoalDtoByGoalId((long) 1).getId());
+    }
+
+    @Test
+    @DisplayName("Test getGoalById(): returns dto of goal by id")
+    void testGetGoalById() {
+        when(goalStore.getGoalById(1L))
+                .thenReturn(java.util.Optional.ofNullable(goal));
+        assertEquals(new Long(1), goalApi.getGoalDtoByGoalId((long) 1).getId());
```

Comment:
Or just `1L` instead of `new Long(1)` or `(long) 1`

Reply:
Done